[2.5.7] Revised addition of GKE operator chart

cmurphy opened the pull request:
This is a revision of #412 that incorporates the feedback from that pull request and updates the CRD to align with the current CRD in https://github.com/rancher/gke-operator/pull/3. The structure and authorship of the author's original commits are not altered, only an additional commit is added.

Klaven committed:
init charts

Signed-off-by: Marek Counts <countsmarek@gmail.com>

Klaven committed:
updated crd

Signed-off-by: Marek Counts <countsmarek@gmail.com>

cmurphy committed:
Fixup addition of GKE operator chart

* Fix the version and variable names
* Update the CRD

cbron commented on charts/rancher-gke-operator/1.0.000/Chart.yaml:
```diff
@@ -0,0 +1,8 @@
+sources:
+  - "https://github.com/rancher/gke-operator"
+version: 1.0.000
+appVersion: v1.0.0
```
Lets start with app version 1.0.1 and version 1.0.100. Only because technically `1.0.000` is invalid semver, so its a weak point in our triple-zero system, even though it likely won't cause any failures.

rmweir replied:
@cbron app version usually just matches the operator version.

cmurphy replied:
Okay so what's the consensus? `version: 1.0.100` and `appVersion: v1.0.0` ?

cbron replied:
Ideally we keep all three together, all at 1.0.1XX

cbron commented on charts/rancher-gke-operator/1.0.000/templates/clusterrole.yaml:
```diff
@@ -0,0 +1,12 @@
+metadata:
+  name: gke-operator
+  namespace: cattle-system
+rules:
```
No secrets in here yet, because no cloud creds yet ? We'll need to remember that.

rmweir replied:
I do think secrets are being read at least already, right?

cmurphy replied:
Not currently but they will be shortly, I can add that back since we know it will be needed

cbron commented on charts/rancher-gke-operator/1.0.000/values.yaml:
```diff
@@ -0,0 +1,11 @@
+gkeOperator:
+  image:
+    repository: rancher/gke-operator
+    tag: v1.0.0
```
Bump this to 1.0.1 to match chart.

rmweir replied:
@cbron we'll probably have rc images first, also why start at 1.0.1?

cbron replied:
Because of the leading zero's being invalid semver. Not supposed to do 000. 
So this tag can match the appVersion in Chart.yaml, and we would just cut rc's on that version until we are done.

rmweir commented on charts/rancher-gke-operator/1.0.000/questions.yaml:
```diff
@@ -0,0 +1 @@
+rancher_min_version: 2.5.7
```
Min should be `2.5.7-rc1`, I believe the `rc1` is arbitrary but it needs some sort of pre-release so it works on those.

cmurphy commented:
Changed the version and appVersion and image tag, also added back the secret perms to the clusterrole including the create verb in light of this comment https://github.com/rancher/gke-operator/pull/3#pullrequestreview-599809746

cbron commented on charts/rancher-gke-operator/1.0.100/questions.yaml:
```diff
@@ -0,0 +1 @@
+rancher_min_version: 2.5.7
```
Hmm I thought this was right before. Should be 2.5.7-alpha1. Ex: https://github.com/rancher/system-charts/blob/dev-v2.5/charts/rancher-eks-operator/1.0.100/questions.yaml

cmurphy replied:
Fixed

thedadams commented on charts/rancher-gke-operator/1.0.100/values.yaml:
```diff
@@ -0,0 +1,11 @@
+gkeOperator:
+  image:
+    repository: rancher/gke-operator
+    tag: v1.0.1
```
```suggestion
    tag: v1.0.1-rc1
```

cmurphy replied:
done

thedadams commented on charts/rancher-gke-operator/1.0.100/Chart.yaml:
```diff
@@ -0,0 +1,8 @@
+sources:
+  - "https://github.com/rancher/gke-operator"
+version: 1.0.100
+appVersion: v1.0.1
```
```suggestion
appVersion: v1.0.1-rc1
```

cmurphy replied:
done

cbron reviewed (commented):
lgtm, waiting for 2.5.6 release

rmweir reviewed (approved):
LGTM
